tests/test_salami.py

```diff
@@ -107,3 +104,86 @@ def test_validate_valid(dataset_path, mocker, mock_validator):
     missing_files, invalid_checksums = salami.validate(dataset_path)
     assert not (missing_files or invalid_checksums)
     mock_validator.assert_called_once()
+
+
+def test_load_track_invalid_track_id():
+    with pytest.raises(ValueError):
+        salami.load_track('a_fake_track')
+
+
+def test_load_track_no_metadata():
+    with pytest.raises(EnvironmentError):
+        salami.SALAMI_METADATA = None
+        salami.load_track('10')
+
+
+def test_load_track_wrong_metadata():
+    with pytest.raises(EnvironmentError):
+        salami.SALAMI_METADATA = {'data_home': 'test'}
+        salami.load_track('10', 'wrong_data_home')
+
+
+def test_load_track_missing_metadata(mocker):
+    data_home = 'fake-data-home'
+    salami.SALAMI_METADATA = {'data_home': data_home}
+    salami.load_sections = mocker.MagicMock(return_value=['a', 'b', 'c', 'd'])
+
+    expected_track = salami.SalamiTrack(
+        '10',
+        'fake-data-home/Salami/audio/10.mp3',
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+        None,
+    )
+
+    actual_track = salami.load_track('10', data_home)
+    assert actual_track == expected_track
+
+
+def test_load_track_with_metadata(mocker):
+    track_metadata = {
+        'source': 'source', 'annotator_1_id': 'aid-1', 'annotator_2_id': 'aid-2',
+        'duration_sec': 60, 'title': 'title', 'artist': 'artist',
+        'annotator_1_time': None, 'annotator_2_time': None, 'class': 'class',
+        'genre': 'genre'
+    }
+    data_home = 'fake-data-home'
+    salami.SALAMI_METADATA = {'data_home': data_home, '10': track_metadata}
+    salami.load_sections = mocker.MagicMock(return_value=['a', 'b', 'c', 'd'])
```

Comment:
Same comment as above. Since you are using this mock twice, you can also do something like this:

```
@pytest.fixture
def mock_load_sections():
  return mocker.patch.object(salami, 'load_sections')

...

def test_load_track_with_metadata(mock_load_sections) :
  ...

   mock_load_sections.return_value = ['a' ... 'e']
```

Reply:
Note that this will also allow you to use a different mock between tests that is correctly scoped and you don't have to worry about things breaking between tests.

Reply:
Done!